pkg/controllers/collaset/synccontrol/update.go

```diff
@@ -416,3 +474,68 @@ func (u *recreatePodUpdater) GetPodUpdateFinishStatus(podInfo *PodUpdateInfo) (f
 	// Recreate policy alway treat Pod as update finished
 	return podInfo.IsUpdatedRevision && !podInfo.PodDecorationChanged, "", nil
 }
+
+type replaceUpdatePodUpdater struct {
+	collaSet *appsv1alpha1.CollaSet
+	ctx      context.Context
+	client.Client
+}
+
+func (u *replaceUpdatePodUpdater) AnalyseAndGetUpdatedPod(updatedRevision *appsv1.ControllerRevision, podUpdateInfo *PodUpdateInfo) (inPlaceUpdateSupport bool, onlyMetadataChanged bool, updatedPod *corev1.Pod, err error) {
+	// when replaceUpdate, inPlaceUpdateSupport and onlyMetadataChanged always false
+
+	// 1. judge replace pair new pod is updated revision, if not, delete.
+	if podUpdateInfo.replacePairNewPodInfo != nil {
+		newPodRevision, exist := podUpdateInfo.replacePairNewPodInfo.Pod.Labels[appsv1.ControllerRevisionHashLabelKey]
+		if exist && newPodRevision == updatedRevision.Name {
+			return
+		}
+		patch := client.RawPatch(types.StrategicMergePatchType, []byte(fmt.Sprintf(`{"metadata":{"labels":{"%s":"%d"}}}`, appsv1alpha1.PodDeletionIndicationLabelKey, time.Now().UnixNano())))
```

Comment:
I think we should refactor the interface `PodUpdater`.

`PodUpdater` should provide a method named like `DoUpdatePod`. Each PodUpdateStrategy should have its own implementation to do actual Pod update. We should move the Pod operations of all PodUpdateStrategy into this method.